libdnf/module/module_sack.cpp

```diff
@@ -149,6 +155,28 @@ BaseWeakPtr ModuleSack::get_base() const {
     return p_impl->base;
 }
 
+
+void ModuleSack::Impl::module_defaults_resolve() {
+    module_metadata.resolve_added_metadata();
+    module_defaults = module_metadata.get_default_streams();
+}
+
+
+const std::string & ModuleSack::get_default_stream(const std::string & name) const {
+    p_impl->module_defaults_resolve();
```

Comment:
This might be a performance problem. For each request the current workflow will go to libmodulemd and requests the resolve defaults for all modules and creates a new object in C++ that is used for name search later on (but only once). May be I overlooked something. If the issue is there anyway we do not need to address it right now but it is possible to create a TODO.